File: scripts/src/create-cname.rs
use config::CONFIG;
use std::fs;
use std::io::Write;
use std::path;

fn main() {
    let cname_asset_path = path::Path::new("app/assets/CNAME");

    if cname_asset_path.exists() {
        fs::remove_file(cname_asset_path).unwrap();
    }
    let mut cname_file = fs::File::create(cname_asset_path).unwrap();

    cname_file
        .write_all(CONFIG.domain.as_bytes())
        .expect("Unable to write CNAME asset file")
}

File: scripts/create-cname.ts
import fs from 'node:fs/promises';

const cnameAssetPath = 'app/public/assets/CNAME';
const cnameAssetExists = await fs
  .access(cnameAssetPath)
  .then(() => true)
  .catch(() => false);
if (cnameAssetExists) {
  await fs.unlink(cnameAssetPath);
}

await fs.writeFile(cnameAssetPath, 'simpleicons.org');
